lib/storage/metricsmetadata/store.go

```diff
@@ -0,0 +1,382 @@
+package metricsmetadata
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/bytesutil"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/fasttime"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/fs"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+type metadataKey struct {
+	accountID        uint32
+	projectID        uint32
+	metricFamilyName string
+}
+
+type metricTimingInfo struct {
+	LastIngestionTime    uint64 `json:"lastIngestionTime"`
+	AvgIngestionInterval uint64 `json:"avgIngestionInterval"`
+	IngestionCount       int64  `json:"ingestionCount"`
+}
+
+type recordForStore struct {
+	AccountID        uint32
+	ProjectID        uint32
+	MetricFamilyName string
+	Rows             []Row
```

Comment:
why both `recordForStore ` and each `Row ` have tenant fields?

Reply:
removing persistence effectively resolves this, so marking as resolved